### DISCUSSION TITLE
[Q] Improve long time process to write file on the disk?

### DISCUSSION THREAD

**Discussion by juanzone** [exceljs/exceljs#2408] [Q&A]
Hi everyone, 

I started using this lib and faced an issue where it took quite some time to save the file on the disk. 

Here are the logs:

```
------START_READ_FILE------ 2020-10-08T17:16:55.890Z
------END_READ_FILE------ 2020-10-08T17:16:57.703Z
------START_WRITE_CELLS------ 2020-10-08T17:16:57.706Z
------END_WRITE_CELLS------ 2020-10-08T17:16:57.707Z
------START_WRITE_FILE_DISK------ 2020-10-08T17:16:57.707Z
------END_WRITE_FILE_DISK------ 2020-10-08T17:17:02.051Z
```
As you can see it took 2 secs to read the file and **5** to write the file on the disk.

Is it normal? How can I improve the writing time? 

The file is 137ko.

I'm using xlsx readFile to read the file
`  const sheet = (await book.xlsx.readFile(inputPath)).worksheets[0];
`

I'm using xlsx writeFile to write the file
`  await book.xlsx.writeFile(outputPath);
`

Thanks in advance :)

**Comment by oscarcoding** [comment]
I also see this problem where some files take a very long time to read and write, eg from my logs

```
[02:30:31 5170]  START readFile .... 
[02:30:41 1680]  ... DONE readFile
[02:30:41 1700]  START writeFile ....
[02:31:24 9620]   ... DONE writeFile
[02:31:24 9630]  Set headers
[02:31:24 9630]  Send file

```
This is nearly 10 seconds to read a 65kb file and nearly 1 minute to write the same file.  It is way too long and I wont be able to continue using exceljs if this is not fixable.

Currently using v4.3 but I also had this problem in v3
